File: apps/admin-v2/src/domain/repositories/auth.repository.ts
import type { User } from '../entities/user';
import type { Tenant } from '../entities/tenant';

export interface LoginResult {
  user: User;
  token: string;
  tenant: Tenant | null;
}

export interface RegisterResult extends LoginResult {
  emailVerified: boolean;
}

export interface AuthRepository {
  login(identifier: string, password: string): Promise<LoginResult>;
  register(data: {
    name: string;
    email: string;
    password: string;
    businessName?: string;
    businessType?: string;
  }): Promise<RegisterResult>;
  verifyEmail(email: string, code: string): Promise<LoginResult>;
  resendVerification(email: string): Promise<void>;
  requestPasswordReset(email: string): Promise<void>;
  resetPassword(input: { email: string; token: string; password: string }): Promise<void>;
  logout(): Promise<void>;
  me(): Promise<{ user: User; tenant: Tenant | null }>;
}

File: apps/admin_mobile/lib/domain/repositories/auth_repository.dart
import '../entities/user.dart';
import '../entities/tenant.dart';

class LoginResult {
  final User user;
  final String token;
  final Tenant? tenant;

  const LoginResult({
    required this.user,
    required this.token,
    this.tenant,
  });
}

abstract class AuthRepository {
  Future<LoginResult> login(String email, String password);
  Future<LoginResult> register({
    required String name,
    required String email,
    required String password,
  });
  Future<void> logout();
  Future<({User user, Tenant? tenant})> me();
}
